src/application_settings/configuring_base.py

```diff
@@ -1,129 +1,30 @@
 """Module for handling configuration."""
-import sys
-from dataclasses import fields
-from pathlib import Path
-from re import sub
 from typing import Any, TypeVar
 
-from pathvalidate import is_valid_filepath
 from pydantic.dataclasses import dataclass
 
-if sys.version_info >= (3, 11):
-    import tomllib
-else:
-    import tomli as tomllib
+from .container_base import ContainerBase, ContainerSectionBase
 
-
-TConfig = TypeVar("TConfig", bound="ConfigBase")  # pylint: disable=invalid-name
-TConfigSection = TypeVar(  # pylint: disable=invalid-name
-    "TConfigSection", bound="ConfigSectionBase"
-)
-
-
-_ALL_CONFIGS: dict[int, Any] = {}
+ConfigT = TypeVar("ConfigT", bound="ConfigBase")
+ConfigSectionT = TypeVar("ConfigSectionT", bound="ConfigSectionBase")
 
 
 @dataclass(frozen=True)
-class ConfigSectionBase:
+class ConfigSectionBase(ContainerSectionBase):
     """Base class for all ConfigSection classes (so that we can bound a TypeVar)"""
 
 
 @dataclass(frozen=True)
-class ConfigBase:
+class ConfigBase(ContainerBase):
     """Base class for main Config class"""
 
     @classmethod
-    def default_config_foldername(cls: type[TConfig]) -> str:
-        """Return the class name without 'Config', lowercase, with a preceding dot and underscores to seperate words."""
-        return (
-            "."
-            + sub("(?<!^)(?=[A-Z])", "_", cls.__name__.replace("Config", "")).lower()
+    def kind_string(cls: type[ConfigT]) -> str:
+        "Return Config"
+        return "Config"
```

Comment:
Is this mend to be user-overrideable or something we define as library? In the second case I would prefer to have something like:

```
@enum.unique
class ContainerType(enum.Enum):
     Config = "Config"
     Settings = "Settings"
```

That way we easily export the different options, and ensure they are unique.

Reply:
This is not meant to be user-overrideable. I decided to introduce a ContainerTypeStr-type, being a literal that is either Config or Settings. That also accomplishes what you want, right?